Fix with a partial revert. "topology: cmake: use local ALSA utilities to build topology"

lgirdwood opened the pull request:
This reverts commit 47dd8d33fc9ead3ee2d3e378a9ad93e889d41e68.

sof-ctl does not need git version of alsa-lib like the topology compiler but can use system version of alsa-lib to build.

kv2019i commented on tools/topology/CMakeLists.txt:
```diff
@@ -1,19 +1,18 @@
 
 function(alsatplg_version  OUT_STATUS  OUT_VERSION)
-	execute_process(COMMAND ${ALSATPLG_CMD} --version
+	execute_process(COMMAND alsatplg --version
```
Commit message refers to alsactl but this reverts the whole thing. This will cause invalid tplgs to be created by our CI system... (as the system alsa-lib is too old).

kv2019i reviewed (commented):
Not sure do we want to revert the tplg part without an alternative solution in place?

lrgirdwo committed:
tools: ctl: use system alsa to build sof-ctl tool

The SOF ctl tool is conservative with API and does not need a bleeding
edge ALSA library so it can be built with system ALSA.

Signed-off-by: Liam Girdwood <liam.r.girdwood@linux.intel.com>

kv2019i reviewed (approved):
Looks good now. Only concern is that this require build host to have system asound-dev installed. I know @cgturner1 had at least one build environment where asound-dev was on purpose not installed to ensure the the locallyl built alsa is used used. maybe @cgturner1 to ack this?

lgirdwood commented:
@lyakh ping for review.

lyakh reviewed (approved):
would be good to have a note explaining the reason for the revert somewhere

wszypelt commented:
SOFCI TEST

lgirdwood commented:
SOFCI TEST

intel-sofci commented:
<!-- summary-comment:test-summary -->
### PR 10073: test results

Run date: 2026-08-31 17:49 UTC

Tested commit: [676c93fa8af9825b1823c9c8b340e34e310d5a06](https://github.com/thesofproject/sof/commit/676c93fa8af9825b1823c9c8b340e34e310d5a06)

![mtl pass rate](https://img.shields.io/badge/mtl-critical_failure-black) ![lnl pass rate](https://img.shields.io/badge/lnl-passed-brightgreen) ![ptl pass rate](https://img.shields.io/badge/ptl-critical_failure-black) ![wcl pass rate](https://img.shields.io/badge/wcl-passed-brightgreen) ![nvl pass rate](https://img.shields.io/badge/nvl-critical_failure-black)
